graphHttpServer/GraphHttpServer.cpp

```diff
@@ -0,0 +1,271 @@
+        result_ = std::to_string(value);
+        return true;
+    }
+    virtual bool process(const bool &value) {
+        result_ = (value ? "true" : "false");
+        return true;
+    }
+    virtual bool process(const float &value) {
+        result_ = std::to_string(value);
+        return true;
+    }
+    virtual bool process(const double &value) {
+        result_ = std::to_string(value);
+        return true;
+    }
+    virtual bool process(const std::string &value) {
+        result_ = EscapeJson(value);
+        return true;
+    }
+
+    std::string json() const { return result_; }
+  private:
+    std::string result_;
+};
+
+void ListNodes(Graph *graph, HttpReply *reply) {
+    reply->text += '[';
+    for (int i = 0; i < graph->numNodes(); ++i) {
+        reply->text += '"' + graph->node(i)->name() + '"';
+        if (i + 1 < graph->numNodes()) {
+            reply->text += ",";
+        }
+    }
+    reply->text += ']';
+}
+
+void ListProperties(PropertyList *list, HttpReply *reply) {
+    if (list == 0) {
+        reply->setNotFound();
+        return;
+    }
+
+    std::ostringstream ss;
+    ss << "[";
+    for (int i = 0; i < list->numProperty(); ++i) {
+        const NamedProperty *property = list->property(i);
+        ToJsonValue converter;
+        property->apply(&converter);
+        ss << "{name:" <<  EscapeJson(property->name())
+          << ",type:" << EscapeJson(property->typeName())
+          << ",value:" << converter.json() << "}";
+
+        if (i + 1 < list->numProperty()) {
+            ss << ",";
+        }
+    }
+    ss << "]";
+
+    reply->text += ss.str();
+}
+
+void ServeNode(NodeBase *node, HttpReply *reply) {
+    if (!node) {
+        reply->setNotFound();
+        return;
+    }
+
+    std::ostringstream ss;
+    ss << "{name:" << EscapeJson(node->name()) << ",output:[";
+
+    for (int i = 0; i < node->numOutputStream(); ++i) {
+        NamedStream *stream = node->outputStream(i);
+        ss << "{name:" << EscapeJson(stream->streamName())
+          << ",type:" << EscapeJson(stream->typeName()) << "}";
+        if (i + 1 < node->numOutputStream()) {
+            ss << ',';
+        }
+    }
+    ss << "],input:[";
+
+    for (int i = 0; i < node->numInputPin(); ++i) {
+        NamedPin *pin = node->inputPin(i);
+        ss << "{name:" << EscapeJson(pin->name())
+          << ",type:" << EscapeJson(pin->typeName());
+        if (pin->isConnected()) {
+            ss << ",connection:{node:"
+              << EscapeJson(pin->connectedStream()->node()->name())
+              << ",stream:" << EscapeJson(pin->connectedStream()->streamName())
+              << "}";
+        }
+        ss << "}";
+        if (i + 1 < node->numInputPin()) {
+            ss << ',';
+        }
+    }
+    ss << "]}";
+    reply->text += ss.str();
+}
+
+// Handle URI with prefix /node/nodeName, assuming nodeName is valid.
+void ServeValidNodeDir(std::vector<std::string> directories, NodeBase *node, HttpReply *reply) {
```

Comment:
[minor] you can pass directories by ref and avoid a copy

Reply:
done